extension/src/documents.ts

```diff
@@ -99,33 +206,44 @@ export class DocumentManager{
 		//		It is important to start tracking this early because we can't actually see what is opened by VSCode during its internal workspace reload.
 		//		Our first opportunity to capture these previously opened documents is when they are activated. **Unavoidable Technical Debt that needs future resolution**
 		if (vscode.window.activeTextEditor && this.getSelectorType(vscode.window.activeTextEditor.document.fileName) === DocumentManager.Selectors.lsp) {
-			this._opened.set(vscode.window.activeTextEditor.document.uri.fsPath, vscode.window.activeTextEditor.document);
+			const key = this.documentConsumeOrValidate(vscode.window.activeTextEditor.document, Origins.OPENED);
+			shouldExist.push(key);
 		}
 
 		// This builds the '_workspace' *.LSP Memory Document placeholder set
 		// 		This feature does make the "fair assumption" that AutoCAD machines have plenty of memory to be holding all this information
 		// 		The impact of creating read-only documents was stress tested with a root workspace folder containing 10mb of *.LSP files
-		//		and the memory footprint from the readonlyDocument's increased the memory (sustained) by less than 50mb		
+		//		and the memory footprint from just the ReadonlyDocument's increased the memory (sustained) by less than 50mb		
 		vscode.workspace.findFiles("**").then((items: vscode.Uri[]) => {
-			items.forEach((fileUri: vscode.Uri) => {			
-				if (this.getSelectorType(fileUri.fsPath) === DocumentManager.Selectors.lsp) {
-					this._workspace.set(fileUri.fsPath, ReadonlyDocument.open(fileUri.fsPath));
+			items.forEach((fileUri: vscode.Uri) => {	
+				const key = this.pathConsumeOrValidate(fileUri.fsPath, Origins.WSPACE);
+				if (key !== '') {
+					shouldExist.push(key);
+				}
+			});
+		});
+
+
+		if (ProjectTreeProvider.hasProjectOpened()) {
+			this.projectKeys.forEach(key => {
+				key = this.pathConsumeOrValidate(key, Origins.PROJECT);
+				if (key !== '') {
+					shouldExist.push(key);
 				}
 			});
+		}
+
+
+		/////////////////////// The extension appears to reset after every workspace change and this probably isn't really doing anything ///////////////////////
+		this.cacheKeys.filter(k => !shouldExist.includes(k)).forEach(obsolete => {
```

Comment:
we also noticed this behavior, the extension will reset if open a workspace, open a folder.

When reset the extension, I think the `cacheKeys` will get lost (or reset) automatically?

Reply:
Yes, the only reason I left it was because this actually does do all the work to reset the state. However, when it is done, it also appears that it completely resets the extension wasting that work. If you don't have any ideas on how to persist that data past the reset, then it can go away.

Reply:
I don't have any ideal to persist the cache between the different folders. If the `cacheKeys` will be also reset automatically, I prefer to make the codes more simpler and cleaner (remove the `shouldExist`)